org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/impl/ChangeSerializer.java

```diff
@@ -95,8 +95,9 @@ else if (context instanceof ResourceSet) {
 		
 		final SubMonitor monitorInAll = SubMonitor.convert(this.monitor,
 				weightedPrepareAndApplyEffort + weightedCreateTextChangesEffort);
-		final SubMonitor monitorPrepareAndApply = monitorInAll.split(weightedPrepareAndApplyEffort, SubMonitor.SUPPRESS_NONE);
-		monitorPrepareAndApply.beginTask("Preparing and applying file changes...", weightedPrepareAndApplyEffort);
+		final SubMonitor monitorPrepareAndApply = SubMonitor.convert(
+				monitorInAll.split(weightedPrepareAndApplyEffort, SubMonitor.SUPPRESS_NONE),
```

Comment:
why do you split here again? should be enough to convert monitorInAll, or what am I missing?

Reply:
I need to split with `suppressflags` set to `SUPPRESS_NONE`, as the default setting made by `split(int)` is `SUPPRESS_BEGINTASK`, which would prevent reporting the task description `"Preparing and applying file changes..."` to the root monitor because of the following check in `SubMonitor.beginTaskImpl()`:
```
if ((flags & SUPPRESS_BEGINTASK) == 0 && name != null)
```
But I want it to be shown in the UI.

This behavior is tested by https://github.com/eclipse/xtext-eclipse/blob/387fbf08c8f34c5f50cae05229be59a8ec004949/org.eclipse.xtext.ui.tests/src-longrunning/org/eclipse/xtext/ui/tests/refactoring/ProgressReportingTest.xtend (xtext-eclipse)